include/deal.II/base/work_stream.h

```diff
@@ -747,29 +747,143 @@ namespace WorkStream
 
         tf::Task last_copier;
 
+        struct CopyChunk
+        {
+          std::vector<CopyData> copy_datas;
+
+          CopyChunk(const unsigned int chunk_size,
+                    const CopyData    &sample_copy_data)
+          {
+            copy_datas.resize(chunk_size, sample_copy_data);
+          }
+        };
+
         // idx is used to connect each worker to its copier as communication
         // between tasks is not supported. It does this by providing a unique
         // index in the vector of pointers copy_datas at which the copy data
         // object where the work done by work task #idx is stored
         unsigned int idx = 0;
 
-        std::vector<std::unique_ptr<CopyData>> copy_datas;
+        std::vector<std::unique_ptr<CopyChunk>> copy_chunks;
 
+        std::vector<Iterator> chunk;
+        chunk.reserve(chunk_size);
+
+        unsigned int chunk_counter = 0;
         // Generate a static task graph. Here we generate a task for each cell
         // that will be worked on. The tasks are not executed until all of them
         // are created, this code runs sequentially.
-        for (Iterator it = begin; it != end; ++it, ++idx)
+        for (Iterator it = begin; it != end; ++it)
           {
-            copy_datas.emplace_back();
+            chunk.emplace_back(it);
+            chunk_counter += 1;
+            if (chunk_counter == chunk_size)
+              {
+                copy_chunks.emplace_back(
+                  std::make_unique<CopyChunk>(chunk_size, sample_copy_data));
```

Comment:
You are still creating `N/chunk_size` copy data objects here, which is quite an expensive operation if you loop over a large collection of things. Is that on your list?

I have to admit that I'm surprised you went this way. The existing TBB implementation already has all of this. It just needed to be copied.

Reply:
This is still an artifact of the static task graph problem like before, each copier task has to have its own copy object defined before the task graph can execute. This problem is solved by a parallel pipeline implementation, which is indeed the next step I wish to tackle. Since chunking needed to be implemented for the colored implementation I figured I would include support for the non-colored version as well. I don't intend this to be a permanent solution.

Reply:
Sorry I see now that there is some avoidable inefficiency here in the allocation of the copy datas internally at this stage in the process. It is unavoidable to have a vector of pointers but I need not initialize what they are pointing to at this stage.